tests/test_odilo.py

```diff
@@ -0,0 +1,250 @@
+    DatabaseTest,
+    sample_data
+)
+
+from core.model import (
+    Collection,
+    ConfigurationSetting,
+    DataSource,
+    DeliveryMechanism,
+    ExternalIntegration,
+    Identifier,
+    Library,
+    LicensePool,
+    Representation,
+    RightsStatus,
+    SessionManager
+)
+
+Base = declarative_base()
+app = Flask(__name__)
+
+testing = True
+# testing = 'TESTING' in os.environ
+db_url = Configuration.database_url(testing)
+# Initialize a new database session unless we were told not to
+# (e.g. because a script already initialized it).
+autoinitialize = os.environ.get('AUTOINITIALIZE') != 'False'
+if autoinitialize:
+    SessionManager.initialize(db_url)
+session_factory = SessionManager.sessionmaker(db_url)
+_db = flask_scoped_session(session_factory, app)
+if autoinitialize:
+    SessionManager.initialize_data(_db)
+
+
+class OdiloAPITest(DatabaseTest, unittest.TestCase):
+    def setup(self):
+        super(OdiloAPITest, self).setup()
+        library = self._default_library
+        self._db = _db
+        self.collection = MockOdiloAPI.mock_collection(self._db)
+        self.circulation = CirculationAPI(self._db, library, api_map={ExternalIntegration.ODILO: MockOdiloAPI})
+        self.api = self.circulation.api_for_collection[self.collection.id]
+
+    def setUp(self):
+        self.setUp_common()
+        self._db = _db
+
+        # Load from DB or fake collection
+        self.collection = None
+        self.fake_collection()
+        self.api = OdiloAPI(self._db, self.collection)
+
+    def fake_collection(self):
+        collection = Collection
+        collection.protocol = ExternalIntegration.ODILO
+        collection.id = 1
+
+        collection.external_integration = ExternalIntegration
+        collection.external_integration.username = 'Odilo'
+        collection.external_integration.password = 'secret'
+
+        # external_integration = collection.create_external_integration(ExternalIntegration.ODILO, self._db)
+
+        collection.external_integration.settings.info = {
+            OdiloAPI.LIBRARY_API_BASE_URL: 'http://localhost:8080/api/v2',
+            'username': 'Odilo',
+            'password': 'secret'
+        }
+
+        self.collection = collection
+
+    @classmethod
+    def setUp_common(cls):
+        print 'setUp common'
+
+    @classmethod
+    def sample_data(cls, filename):
+        return sample_data(filename, 'odilo')
+
+    @classmethod
+    def sample_json(cls, filename):
+        data = cls.sample_data(filename)
+        return data, json.loads(data)
+
+    def error_message(self, error_code, message=None, token=None):
+        """Create a JSON document that simulates the message served by
+        Odilo given a certain error condition.
+        """
+        message = message or self._str
+        token = token or self._str
+        data = dict(errorCode=error_code, message=message, token=token)
+        return json.dumps(data)
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
I think this answers my 'how are you running the tests' question. We run tests with `nosetests` so this code is unnecessary.